src/app/g3w-ol/controls/basequerypolygoncontrol.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * @file
+ * @since v3.8
+ */
+import { SPATIALMETHODS, VM } from 'g3w-ol/constants';
+import GUI from 'services/gui';
+
+const { merge }                  = require('core/utils/ol');
+const InteractionControl         = require('g3w-ol/controls/interactioncontrol');
+
+// TODO: make it easier to understand.. (what variables are declared? which ones are aliased?)
+const {
+  Geometry : {
+    getAllPolygonGeometryTypes,
+    isPolygonGeometryType
+  }
+} = require('core/utils/geo');
+
+const VALIDGEOMETRIES = getAllPolygonGeometryTypes();
+
+const BaseQueryPolygonControl = function(options = {}) {
+
+  const {
+    spatialMethod=SPATIALMETHODS[0],
+    name,
+    tipLabel,
+    label,
+    onSelectlayer,
+    enabled=true,
+    interactionClass
+  } = options;
+  this.layers = options.layers || [];
+
+  /**
+   * @type {unknown[]}
+   *
+   * @since 3.8.0
+   */
+  this.externalLayers = [];
+
+  this.unwatches = [];
+
+  this.listenLayersVisibilityChange();
+
+  options.visible = this.checkVisibile(this.layers);
+
+  /**
+   * Store current selected layer
+   *
+   * @since 3.8.0
+   */
+  this.selectedLayer = null;
+
+  const _options = {
+    offline: false,
+    name,
+    tipLabel,
+    label,
+    // update selected layer
+    onSelectlayer,
+    clickmap: true, // set ClickMap
+    interactionClass,
+    spatialMethod,
+    toggledTool:{
+      type: 'spatialMethod',
+      how: 'toggled' // or hover
+    },
+    onhover: true
+  };
+  options = merge(options,_options);
+  options.geometryTypes = VALIDGEOMETRIES;
+  InteractionControl.call(this, options);
+  //starting disabled
+  this.setEnable(enabled);
+
+  this._handleExternalLayers();
+};
+
+ol.inherits(BaseQueryPolygonControl, InteractionControl);
+
+const proto = BaseQueryPolygonControl.prototype;
+
+/**
+ * @param { unknown | null } layer
+ *
+ * @since 3.8.0
+ */
+
+proto.setSelectedLayer = function(layer) {
+  this.selectedLayer = layer;
+};
+
+proto.change = function(layers=[]){
+  this.layers = layers;
+  const visible = this.checkVisibile(layers);
+  this.setVisible(visible);
+  this.setEnable(false);
+  this.listenLayersVisibilityChange();
+};
```

Comment:
check if the `change()` method is overrides/implements another one from a parent class (e.g. `InteractionControl`) or is native to `BaseQueryPolygonControl`